android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUITabLayoutTabGroup.java

```diff
@@ -48,6 +48,31 @@ public TiUITabLayoutTabGroup(TabGroupProxy proxy, TiBaseActivity activity)
 		super(proxy, activity);
 	}
 
+	private void setDrawablesForTab(int tabIndex)
+	{
+		TabProxy tabProxy = ((TabGroupProxy) this.proxy).getTabList().get(tabIndex);
```

Comment:
We need to validate the given `tabIndex` before using it. This is especially needed if the `TabGroup` has no tabs. I've seen some app devs add/remove tabs dynamically before.
```java
ArrayList<TabProxy> tabProxiesList = ((TabGroupProxy) this.proxy).getTabList();
if ((tabIndex < 0) || (tabIndex >= tabProxiesList.length())) {
	return;
}

TabProxy tabProxy = tabProxiesList.get(tabIndex);
if (tabProxy == null) {
	return;
}
```

Reply:
We should test this with an empty `TabGroup` just in case.
```javascript
var tabGroup = Ti.UI.createTabGroup({
	tabsBackgroundColor: "red",
	tabsBackgroundSelectedColor: "orange",
});
tabGroup.tabsBackgroundColor = "green";
tabGroup.tabsBackgroundSelectedColor = "yellow";
tabGroup.addEventListener("open", function() {
	tabGroup.tabsBackgroundColor = "purple";
	tabGroup.tabsBackgroundSelectedColor = "blue";
});
tabGroup.open();
```